keanu-project/src/main/java/io/improbable/keanu/vertices/ProposalVertex.java

```diff
@@ -0,0 +1,7 @@
+package io.improbable.keanu.vertices;
+
+import java.util.List;
+
+public interface ProposalVertex {
```

Comment:
Unless I've misunderstood something, it feels like it would be better to parametrise this class by `<T extends Tensor>`, and expect `List<T>` instead of `List<?>` (where the `T` will be the same type that the concrete Vertex class is parametrised on). 

That avoids the need for a yucky cast to `DoubleTensor` in your implementation of the method in `GaussianVertex`.